src/qt3utils/applications/qt3ple/main.py

```diff
@@ -0,0 +1,503 @@
+        grid = plt.GridSpec(2, num_readers)
+        self.update_image(model, grid)
+        self.update_plot(model, grid)
+
+    def update_image(self, model, grid) -> None:
+        for ii, reader in enumerate(model.readers):
+
+            #if self.log_data:
+            #    data = np.log10(model.scanned_count_rate)
+            #    data[np.isinf(data)] = 0  # protect against +-inf
+            #else:
+            #    data = model.scanned_count_rate
+            #data = np.array(data).T.tolist()
+            y_data = []
+            for output in model.outputs:
+                y_scan = output[reader]
+                if isinstance(y_scan[0], list) or isinstance(y_scan[0], np.ndarray):
+                    for jj, ydatum in enumerate(y_scan):
+                        y_scan[jj] = np.mean(np.array(ydatum))
+                y_data.append(y_scan)
+
+            ax = self.fig.add_subplot(grid[0, ii])
+            y_data = np.array(y_data).T
+            artist = ax.imshow(y_data, cmap=self.cmap
+                                    , extent=[model.current_frame + model.wavelength_controller.settling_time_in_seconds,
+                                                                       0.0,
+                                                                       model.get_start,
+                                                                       model.get_end + model.step_size])
+            cbar = self.fig.colorbar(artist, ax=ax)
+
+            #if self.log_data is False:
+            #    cbar.formatter.set_powerlimits((0, 3))
+
+            self.ax.set_xlabel('Pixels')
+            self.ax.set_ylabel('Voltage (V)')
+
+    def update_plot(self, model, grid) -> None:
+
+        for ii, reader in enumerate(model.readers):
+            y_data = model.outputs[model.current_frame-1][reader]
+            x_control = model.scanned_control[model.current_frame-1]
+            ax = self.fig.add_subplot(grid[1, ii])
+            ax.plot(x_control, y_data, color='k', linewidth=1.5)
+
+    def reset(self) -> None:
+        self.ax.cla()
+
+    def set_onclick_callback(self, f) -> None:
+        self.onclick_callback = f
+
+    def onclick(self, event) -> None:
+        pass
+
+
+class SidePanel():
+    def __init__(self, root, scan_range) -> None:
+        frame = tk.Frame(root.root)
+        frame.pack(side=tk.LEFT, fill=tk.BOTH, expand=True)
+
+        row = 0
+        tk.Label(frame, text="Scan Settings", font='Helvetica 16').grid(row=row, column=0, pady=10)
+        row += 1
+        self.start_button = tk.Button(frame, text="Start Scan")
+        self.start_button.grid(row=row, column=0)
+        self.stop_button = tk.Button(frame, text="Stop Scan")
+        self.stop_button.grid(row=row, column=1)
+        self.save_scan_button = tk.Button(frame, text="Save Scan")
+        self.save_scan_button.grid(row=row, column=2)
+        row += 1
+        tk.Label(frame, text="Voltage Range (V)").grid(row=row, column=0)
```

Comment:
This is not always going to be the voltage range, right?

Reply:
This is true. What do you think is a more generic term for it?

Reply:
maybe we can use scanning parameter?